fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexScrubberTest.java

```diff
@@ -568,9 +573,13 @@ void testDetectDanglingFromJoinedIndex() {
                         .setAllowRepair(false)
                         .build())
                 .build()) {
-            assertThrows(RecordDoesNotExistException.class, indexScrubber::scrubDanglingIndexEntries);
+            indexScrubber.scrubDanglingIndexEntries();
             indexScrubber.scrubMissingIndexEntries();
         }
+        Assertions.assertTrue(0 < timer.getCount(FDBStoreTimer.Counts.ONLINE_INDEX_BUILDER_RECORDS_SCANNED));
+        assertEquals(0, timer.getCount(FDBStoreTimer.Counts.ONLINE_INDEX_BUILDER_RECORDS_INDEXED));
+        Assertions.assertTrue(0 < timer.getCount(FDBStoreTimer.Counts.INDEX_SCRUBBER_DANGLING_ENTRIES));
+        assertEquals(0, timer.getCount(FDBStoreTimer.Counts.INDEX_SCRUBBER_MISSING_ENTRIES));
```

Comment:
In theory, we could do with an additional test here that validates the behavior if there's a join index and one (but not all) of the constituents is deleted. I believe, as written, it will identify those as dangling (and I think that's right)